trpc/router/goalV2.ts

```diff
@@ -160,8 +160,17 @@ export const goal = router({
                 offset,
                 ...ctx.session.user,
             });
+            const { starred, watching } = { ...goalsQuery };
 
-            const goals = await query.$castTo<Goal>().execute();
+            let goals = await query.$castTo<Goal>().execute();
+
+            if (starred) {
```

Comment:
This is bad idea, cause you may get is empty list after filtering results 

In this case needs add additional where statement in query for goal's list